src/reducers/VaccineReducer.js

```diff
@@ -0,0 +1,39 @@
+import { VACCINE_ACTIONS } from '../actions/VaccineActions';
+
+const initialState = () => ({
+  scannedSensors: [],
+  isScanning: false,
+});
+
+export const VaccineReducer = (state = initialState(), action) => {
+  const { type } = action;
+
+  switch (type) {
+    case VACCINE_ACTIONS.SCAN_START: {
+      return {
+        ...state,
+        isScanning: true,
+      };
+    }
+    case VACCINE_ACTIONS.SCAN_STOP: {
+      return {
+        ...state,
+        isScanning: false,
+      };
+    }
+    case VACCINE_ACTIONS.SENSOR_FOUND: {
+      const { payload } = action;
+      const { macAddress } = payload;
+
+      const { scannedSensors } = state;
```

Comment:
this may be ok, but I think it's sailing close to the wind - in that it is almost mutating an object in state.
better to use `[...scannedSensors, macAddress]` though I would go slightly further and only update if there's a new one added 😃

Reply:
yeah `push` mutates the array which will mean there won't be any diff and the UI won't re-render. 


[ I suspect this is copied from `ColdChain` which uses https://github.com/immerjs/immer - which is a library which helps you update state by using a proxy of your state object which you update mutably, but causes immutable updates to the real state ]

Reply:
nice tip re `immer` thanks @joshxg I'd noticed that earlier and hadn't made the connection

Reply:
Good to know, thanks @mark-prins for the suggestion and @joshxg for the link! 4ee6470a